x-pack/legacy/plugins/security/index.js

```diff
@@ -128,17 +129,18 @@ export const security = (kibana) => new kibana.Plugin({
       throw new Error('New Platform XPack Security plugin is not available.');
     }
 
+    const config = server.config();
     const xpackMainPlugin = server.plugins.xpack_main;
     const xpackInfo = xpackMainPlugin.info;
     securityPlugin.registerLegacyAPI({
       xpackInfo,
       isSystemAPIRequest: server.plugins.kibana.systemApi.isSystemApiRequest.bind(
         server.plugins.kibana.systemApi
       ),
+      cspRules: createCSPRuleString(config.get('csp.rules')),
```

Comment:
__note:__ if we still need this we'll include that into NP blockers for Security plugin.

Reply:
Included as blocker, for now we agreed with Platform to provide this through legacy API shim and hopefully get support for "lightweight" apps by 8.0 so that we don't need to set CSP manually anymore.